src/main/flight/gps_rescue.c

```diff
@@ -492,14 +476,14 @@ static void performSanityChecks()
     previousTimeUs = currentTimeUs;
 
     if (rescueState.phase == RESCUE_FLY_HOME) {
-        secondsStalled = constrain(secondsStalled + ((rescueState.sensor.velocityToHomeCmS < (0.5 * rescueState.intent.targetVelocityCmS)) ? 1 : -1), 0, 20);
-        if (secondsStalled == 20) {
+        rescueState.intent.secondsFailing = constrain(rescueState.intent.secondsFailing + ((rescueState.sensor.velocityToHomeCmS < (0.5 * rescueState.intent.targetVelocityCmS)) ? 1 : -1), 0, 20);
```

Comment:
```suggestion
        rescueState.intent.secondsFailing += (rescueState.sensor.velocityToHomeCmS < 0.5 * rescueState.intent.targetVelocityCmS) ? 1 : -1;
        rescueState.intent.secondsFailing = constrain(rescueState.intent.secondsFailing, 0, 20);
```

Just a suggestion to make this line more readable imo and split it over two instead.

Reply:
@KarateBrot - thank you, yes, thank you that's much tidier.